crates/primitives/src/transaction/mod.rs

```diff
@@ -280,6 +291,33 @@ pub struct TxEip4844 {
 }
 
 impl TxEip4844 {
+    /// Returns the effective gas price for the given `base_fee`.
+    ///
+    /// Note: this is the same as [TxEip1559::effective_gas_price].
+    pub fn effective_gas_price(&self, base_fee: Option<u64>) -> u128 {
+        match base_fee {
+            None => self.max_fee_per_gas,
+            Some(base_fee) => {
+                // if the tip is greater than the max priority fee per gas, set it to the max
+                // priority fee per gas + base fee
+                let tip = self.max_fee_per_gas - base_fee as u128;
```

Comment:
is this always safe? would we mostly call this fn in blobpool **after** validation?

Reply:
oh, actually no, this should be saturated sub

will fix separately, because this also affects eip1559

Reply:
just added saturating sub